From daad8a5e5ea2dac503513f6ab114184dd427db01 Mon Sep 17 00:00:00 2001
From: Christian Ruppert <idl0r@gentoo.org>
Date: Fri, 2 Apr 2010 16:44:14 +0200
Subject: [PATCH] Fix rsync mirror selectection

Append /gentoo-portage to the rsync mirror url if necessary, bug 312753,
	   thanks to Donald Wallace Rouse II <gentoo-user-0d329vjsd8@dwr2.org>
---
 main.py |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/main.py b/main.py
index 367e8d4..dd3386e 100755
--- a/main.py
+++ b/main.py
@@ -608,6 +608,8 @@ class Interactive(object):
 
 		for (url, args) in sorted(hosts, key = lambda x: (x[1]['country'].lower(), x[1]['name'].lower()) ):
 			marker = ""
+			if options.rsync and not url.endswith("/gentoo-portage"):
+				url+="/gentoo-portage"
 			if (not options.ipv6 and not options.ipv4) and args['ipv6'] == 'y':
 				marker = "* "
 			if options.ipv6 and ( args['ipv6'] == 'n' ): continue
-- 
1.7.0.4

